src/libponyc/pass/refer.c

```diff
@@ -66,6 +66,121 @@ static bool is_this_incomplete(pass_opt_t* opt, ast_t* ast)
   return false;
 }
 
+static const char* generate_multi_dot_name(ast_t* ast, ast_t** def_found) {
```

Comment:
I don't really understand the purpose of this function or how/why it is used - could you add a comment above describing what it does, why it does it, and giving an example of what the output looks like?

Reply:
This function generates the fully qualified string (without the `this`) for a reference (i.e. `a.b.c.d.e`) and ensures it is part of the compiler `stringtab`. It is used to generate this fully qualified string for the field being confused to tracking its `consume`d status via the ast `symtab`. It is also used to generate the fully qualified string to ensure that same field is being assigned to as part of the same expression as the consume. Lastly, it is used to get the definition for the type based on the `ast_data` to ensure at some point the field is tied to a real type even if we haven't quite fully determined the type of each field/subfield reference yet.